int8 id_n1
int8 id_n2